File: Programmers/Swift/가장 큰 정사각형 찾기.swift
import Foundation

var mx = 0
var mp = [[Int]]()

func solution(_ board:[[Int]]) -> Int {
    
    let ySz = board.count
    let xSz = board[0].count
    mp = board
    
    for y in 0..<ySz {
        for x in 0..<xSz {
            if mx<mp[y][x] {
                mx=mp[y][x]
            }
        }
    }
    
    for y in 1..<ySz {
        for x in 1..<xSz {
            if mp[y][x]==0 {continue}
            mp[y][x] = min(mp[y-1][x-1],min(mp[y][x-1],mp[y-1][x]))+1
            if mx<mp[y][x] {mx = mp[y][x]}
        }
    }

    return mx*mx
}

File: Programmers/C++/가장 큰 정사각형 찾기.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#define FOR(i,n) for(int i=0;i<(n);++i)
#define FORS(i,s,n) for(int i=(s);i<(n);++i)
using namespace std;

int mx;

int solution(vector<vector<int>> board) {
    int ySz = board.size();
    int xSz = board[0].size();
    FOR(y,ySz) FOR(x,xSz) mx=max(mx,board[y][x]);
    FORS(y,1,ySz) FORS(x,1,xSz) {
        if(board[y][x]!=0) {
            board[y][x] = min({board[y-1][x-1],board[y-1][x],board[y][x-1]})+1;
            mx = max(mx,board[y][x]);
        }
    }
    
    return mx*mx;
}
